AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -156,6 +144,19 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
 
     // MARK: Private
 
+    private func onDeleteCompletion<M: Model>(result: DataStoreResult<M?>,
+                                              completion: @escaping DataStoreCallback<Void>) {
+        switch result {
+        case .success(let modelOptional):
+            if let model = modelOptional {
+                publishMutationEvent(from: model, mutationType: .delete)
+            }
+            completion(.emptyResult)
+        case .failure(let error):
+            completion(.failure(error))
+        }
```

Comment:
We were previously just calling `completion(result)` -- curious as to why we changed it to completion(.emptyResult) on the success case?

Reply:
because the delete storage behavior now returns an optional model. the Delete first queries for the data, then deletes it. it may result a successful and nothing deleted case. the developer facing API still returns void, since the change in interface resulted in no longer a passthrough from `result` to `completion`, we take the optional model, if delete was successful, model retrieved from modelOptional, then we publish to subscribers. and return .emptyResult for the caller